Fix unhandled exception when RuntimeList.xml contains duplicates.

peterhuene opened the pull request:
This PR fixes an unhandled exception that results when RuntimeList.xml
contains entries that have the same destination subpath when resolving the
runtime pack assets.  An error is now emitted instead of the unhandled
exception.

Additionally, this fixes an unhandled exception that occurs when the
RuntimeList.xml file is not present in the runtime pack that is caused by
the exception message not having a NETSDK prefix.

Also removed an unnecessary override of the `Execute` method for
`GenerateDepsFile` task.  This override should have been removed with a
previous cleanup PR, but was overlooked.

Fixes dotnet/cli#11705.

Commit:
Fix unhandled exception when RuntimeList.xml contains duplicates.

This commit fixes an unhandled exception that results when RuntimeList.xml
contains entries that have the same destination subpath when resolving the
runtime pack assets.  An error is now emitted instead of the unhandled
exception.

Additionally, this fixes an unhandled exception that occurs when the
RuntimeList.xml file is not present in the runtime pack that is caused by
the exception message not having a NETSDK prefix.

Also removed an unnecessary override of the `Execute` method for
`GenerateDepsFile` task.  This override should have been removed with a
previous cleanup PR, but was overlooked.

Fixes dotnet/cli#11705.

peterhuene commented on src/Tasks/Common/Resources/Strings.resx:
```diff
@@ -525,13 +525,21 @@ The following are names of parameters or literal values and should not be transl
-    <value>NETSDK1106: Microsoft.NET.Sdk.WindowsDesktop requires 'UseWpf' or 'UseWindowsForms' to be set to 'true'</value>
-    <comment>{StrBegin="NETSDK1106: "}</comment>
+  <data name="RuntimeListNotFound" xml:space="preserve">
+    <value>NETSDK1108: Runtime list file '{0}' was not found. Report this error to the .NET Core team here: https://github.com/dotnet/core-setup/issues.</value>
```
I'm definitely open to suggestions regarding these two error messages.

dsplaisted replied:
I think there was some discussion in the past about whether we wanted to have links to our repos for filing issues in our error messages. @nguerrera

peterhuene replied:
This is primarily why I wanted suggestions 😄.  These are internal errors that are not actionable by users directly and I'd love to point them towards a way to inform us of the problem, even if it's really unlikely users would ever run into these before we do.  But I can understand not wanting repo references in the error messages.

nguerrera replied:
We could make an aka.ms/report-runtime-list-issue or something?

I don't actually recall such a discussion at this point. cc @livarcocc

peterhuene replied:
Maybe a more generic `aka.ms/dotnet-core-issue` that takes them to `dotnet/core` and we can triage from there?

nguerrera replied:
That works for me.

peterhuene replied:
What about using `aka.ms/dotnet-core-sdk-issue` and we can point it at either the CLI or SDK repos? It's likely that the issue really belongs to core-setup for these two errors, but I think we should be responsible for triage given the errors are coming from the SDK.

We can then use the aka.ms link for any other internal SDK errors in the future.

nguerrera replied:
Ship it.

Commit:
Change error message link to aka.ms.

This commit changes the link in the error messages to an aka.ms link.

nguerrera commented on src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs:
```diff
@@ -85,7 +85,9 @@ protected override void ExecuteCore()
             string runtimeListPath, ITaskItem runtimePack)
-        {            
+        {
+            var assetSubPaths = new HashSet<string>();
```
Should this be OrdinalIgnoreCase?

peterhuene replied:
It should be; even if the file system is case-sensitive, we wouldn't ever want case-insensitive duplicates in the list.

I'll fix.

Commit:
Make asset sub-path set case-insensitive.

This commit makes the hash set that is tracking the asset sub-paths resolved
from runtime packs case-insensitive.

peterhuene commented:
Waiting on some feedback from @KathleenDollard before merging.
